libs/wallet/src/connectors/snap-connector.ts

```diff
@@ -1,247 +1,282 @@
+      const res = await this.invokeSnap<{
+        transactionHash: string;
+        transaction: { signature: { value: string } };
+        receivedAt: string;
+        sentAt: string;
+      }>(JsonRpcMethod.SendTransaction, {
+        publicKey: params.publicKey,
+        sendingMode: params.sendingMode,
+        transaction: params.transaction,
+        networkEndpoints: [this.node],
+      });
+
+      return {
+        transactionHash: res.transactionHash,
+        signature: res.transaction.signature.value,
+        receivedAt: res.receivedAt,
+        sentAt: res.sentAt,
+      };
     } catch (err) {
-      throw SnapConnectorErrors.TRANSACTION_PARSE;
+      if (err instanceof ConnectorError) {
+        throw err;
+      }
+
+      throw ConnectorErrors.sendTransaction;
     }
+  }
 
-    const payload = {
-      method: 'client.send_transaction',
-      params: {
-        sendingMode: 'TYPE_SYNC',
-        transaction: txData,
-        publicKey: pubKey,
-        networkEndpoints: [this.nodeAddress],
-      },
-    };
+  on(event: VegaWalletEvent, callback: () => void) {
+    this.pollListeners[event].push(callback);
+  }
 
-    const result = await invokeSnap<SendTransactionResponse>(
-      this.snapId,
-      payload
+  off(event: VegaWalletEvent, callback?: () => void) {
+    this.pollListeners[event] = this.pollListeners[event].filter(
+      (cb) => cb !== callback
     );
+  }
+  ////////////////////////////////////
+  // Snap methods
+  ////////////////////////////////////
 
-    if ('error' in result) {
-      const { message, code, data } = result.error;
-      throw new WalletError(
-        message,
-        code,
-        typeof data === 'string' ? data : ''
-      );
-    }
+  emit(event: VegaWalletEvent) {
+    this.pollListeners[event].forEach((listener) => {
+      listener();
+    });
+  }
+
+  startPoll() {
+    // This only event we need to poll for right now is client.disconnect,
+    // if more events get added we will need more logic here
+    this.pollRef = setInterval(async () => {
+      const result = await this.isConnected();
+      if (result.connected) return;
+      this.emit('client.disconnected');
+    }, 2000);
+  }
 
-    if (!result?.transaction?.signature) {
-      throw new Error('could not retrieve transaction siganture');
+  stopPoll() {
+    if (this.pollRef) {
+      clearInterval(this.pollRef);
     }
+  }
 
-    return {
-      transactionHash: result.transactionHash,
-      signature: result?.transaction?.signature?.value,
-      receivedAt: result.receivedAt,
-      sentAt: result.sentAt,
-    };
+  /**
+   * Requests permission for a website to communicate with the specified snaps
+   * and attempts to install them if they're not already installed.
+   * If the installation of any snap fails, returns the error that caused the failure.
+   * More informations here: https://docs.metamask.io/snaps/reference/rpc-api/#wallet_requestsnaps
+   */
+  async requestSnap() {
```

Comment:
These utility methods appear to only be used internally to this class.

Suggest making all these methods private so this implementation only exposes public methods that conform to the interface.